test/swe2d/test_thacker.py

```diff
@@ -0,0 +1,60 @@
+# Test for wetting-drying scheme functionality, using Thacker test case
+# Test case details in Gourgue et al (2009)
+
+from thetis import *
+import pytest
+import math
+
+@pytest.mark.parametrize("n,dt,alpha,max_err", [(25, 300., 4., 0.009), (10, 600., 8., 0.06)])
+def test_thacker(n, dt, alpha, max_err):
+    # Mesh
+    mesh2d = SquareMesh(n, n, 951646.46)
+
+    # Bathymetry and initial condition parameters
+    D0 = 50.
+    L = 430620.
+    eta0 = 2.
+    A = ((D0+eta0)**2-D0**2)/((D0+eta0)**2+D0**2)
+    X0 = 475823.23 # Domain offset
+    Y0 = 475823.23
+
+    # Bathymetry
+    bathymetry = Function(FunctionSpace(mesh2d, "CG", 1), name='bathymetry')
+    x = SpatialCoordinate(mesh2d)
+    bathymetry.interpolate(Expression(D0*(1-((x[0]-X0)*(x[0]-X0)+(x[1]-Y0)*(x[1]-Y0))/(L*L))))
+
+    # Solver
+    solverObj = solver2d.FlowSolver2d(mesh2d, bathymetry)
+    options = solverObj.options
+
+    options.dt = dt
+    options.t_end = 43200 - 0.1*options.dt
+    options.t_export = options.dt
+    options.no_exports = True
+    options.timestepper_type = 'cranknicolson'
+    options.shallow_water_theta = 0.5
+    options.wd_alpha = alpha
+    options.solver_parameters_sw = {
+        'snes_type': 'newtonls',
+        'snes_monitor': True,
+        'ksp_type': 'gmres',
+        'pc_type': 'fieldsplit',
+    }
+
+    # Initial conditions
+    x = SpatialCoordinate(mesh2d)
+    elev_init = D0*(sqrt(1-A*A)/(1-A) - 1 - ((x[0]-X0)*(x[0]-X0)+(x[1]-Y0)*(x[1]-Y0))*((1+A)/(1-A)-1)/(L*L))
+    solverObj.assign_initial_conditions(uv=Constant((1e-7,0.)), elev=elev_init)

```

Comment:
Ah, this was to avoid issues with the drag term when velocity is zero. Not required here, as this case doesn't use a drag term.